File: 841/A.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    vector<int> colors(26,0);
    int n;
    cin>>n;
    int friends=0;
    cin>>friends;
    string s;
    cin>>s;
    long long totalcount=0;
    locale loc;
    tolower(s,loc);
    for(int i=0;i<s.length();i++){
        colors[s[i]-'a']++;
        totalcount++;
    }
    if(totalcount<friends){
        cout<<"NO";
        return 0;
    }
    for(int i=0;i<26;i++){
        if(colors[i]>friends){
            cout<<"NO";
            return 0;
        }
    }
    cout<<"YES";
    return 0;
}

File: 841/A.java
import java.io.*;
import java.util.Scanner;
public class A{
    public static void main(String[] args){
        int[] colors= new int[26];
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int friends = s.nextInt();
        int totalcount=0;
        String str;
        s.nextLine();
        str = s.nextLine();
        for(int i=0;i<n;i++){
        colors[str.charAt(i)-'a']++;
        totalcount++;
        }
        for(int i=0;i<26;i++){
            if(colors[i]>friends){
                System.out.println("NO");
                return ;
            }
        }
        System.out.println("YES");
    }
}